langchain/tools/shell/tool.py

```diff
@@ -66,18 +65,16 @@ class ShellTool(BaseTool):
 
     def _run(
         self,
-        commands: List[str],
+        commands: Union[str, List[str]],
         run_manager: Optional[CallbackManagerForToolRun] = None,
     ) -> str:
         """Run commands and return final output."""
         return self.process.run(commands)
 
     async def _arun(
         self,
-        commands: List[str],
+        commands: Union[str, List[str]],
         run_manager: Optional[AsyncCallbackManagerForToolRun] = None,
     ) -> str:
         """Run commands asynchronously and return final output."""
-        return await asyncio.get_event_loop().run_in_executor(
-            None, self.process.run, commands
-        )
+        raise NotImplementedError("Async not implemented for shell tool.")
```

Comment:
how concerned are we about backwards compat? maybe we should check if it's just a string and only error if its a list?

Reply:
Hm actually I think we can keep the get_event_loop logic right now and jsut add a warning.

For backwards compat - since we're relaxing the typing constraints, it doesn't break anything. Both work in the shell right now.

Reply:
as in un-implementing _arun is breaking, no?

Reply:
if someone was previously using

Reply:
Ya I'm gonna not un-implmenet. Misinterpreted something from Ankush

Reply:
ununimplement, as they say

Reply:
🍵 🍵 🍵